src/Servers/Kestrel/Core/src/KestrelServer.cs

```diff
@@ -122,6 +123,12 @@ public async Task StartAsync<TContext>(IHttpApplication<TContext> application, C
 
                 async Task OnBind(ListenOptions options)
                 {
+                    // TODO Why?
+                    if (options.ConnectionAdapters.Any(c => c.IsHttps))
```

Comment:
Also this, I don't think we need to check any connection adapters now.

Reply:
Nuke it.